### PR TITLE
Use summary list component

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
In the DAC report, a number of cases were reported on the change links not being correctly rendered (i.e. without a space between the visible and hidden text). Using the component means we avoid this happening in the future. (It’s also nicer and helps prevent me from continuously flinching…)

### CODE REVIEW

File: app/views/form/check_answers.html.erb [hunk 1]
```diff
@@ -19,11 +19,15 @@
     <% end %>
     <%= display_answered_questions_summary(subsection, @case_log) %>
 
-    <dl class="govuk-summary-list">
+    <%= govuk_summary_list do |summary_list| %>
```
Comment (baarkerlounger) [inline app/views/form/check_answers.html.erb]: This shouldn't have an `=` should it?

File: app/views/form/review.html.erb [hunk 2]
```diff
@@ -21,14 +21,18 @@
             <h3 class="x-govuk-summary-card__title"><%= subsection.label %></h3>
           </div>
           <div class="x-govuk-summary-card__body">
-            <dl class="govuk-summary-list">
+            <%= govuk_summary_list do |summary_list| %>
```
Comment (baarkerlounger) [inline app/views/form/review.html.erb]: Same comment here. This also looks similar enough to above to potentially be extracted into a partial?

File: app/views/form/check_answers.html.erb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/views/form/check_answers.html.erb]: The initial wrapping function should, according to the docs: https://govuk-components.netlify.app/components/summary-list/#input-erb-a-summary-list-with-various-entries

**Review by baarkerlounger — APPROVED** [review]

File: app/views/form/review.html.erb (hunk 2 above)
Comment (paulrobertlloyd) [reply app/views/form/review.html.erb]: > This also looks similar enough to above to potentially be extracted into a partial?

Ah, good catch. Am now a using a partial for the entire list, which means passing around fewer params. 👌

**Pull request merged**